### PR TITLE
refactor(barrier): decouple barrier collect and sync in local barrier manager

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously, the `ManagedBarrierState` in local barrier manager handles the work the both barrier collection and barrier sync. It starts syncing a new barrier immediately after collecting it from all actors. It works in per partial graph manner, which means different partial graphs should always be synced separately. However, in partial checkpoint, we may still collect barrier separately in different partial graphs, but we can include multiple partial graphs in a single commit, and therefore multiple partial graphs can be synced together. Therefore, in this PR, we will first decouple the barrier collect and sync in local barrier manager. The `ManagedBarrierState` only handles the barrier collection. The barrier sync is handled in local barrier worker. 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by wenym1** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19393?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#19393** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19393?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#19383** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19383?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19386** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19386?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wenym1 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/stream/src/task/barrier_manager/managed_state.rs [hunk 1]
```diff
@@ -717,40 +688,29 @@ impl PartialGraphManagedBarrierState {
     /// This method is called when barrier state is modified in either `Issued` or `Stashed`
     /// to transform the state to `AllCollected` and start state store `sync` when the barrier
     /// has been collected from all actors for an `Issued` barrier.
```
Comment (hzxa21) [inline src/stream/src/task/barrier_manager/managed_state.rs:690]: The documentation is outdated given that `ManagedBarrierStateInner` is removed.

**Review by hzxa21 — APPROVED** [review]
Generally LGTM. What is the motivation?

**Comment by wenym1** [comment]
> Generally LGTM. What is the motivation?

Motivation is added to the PR description.

Since the total change for this feature is quite large, will merge this reviewed part first in this PR. Will submit a PR for changes in meta global barrier manager separately.

**Pull request merged**